### ISSUE TITLE
Need syntax to store a temp array

### ISSUE THREAD

**Issue by jrivers96** [Paradigm4/SciDB-Py#122]
We can potentially add something such as the below? "temp=True"?
``
db.build('<x:int8 not null>[i=0:2]', 'i + 10').store('foo', temp=True)
``

**Comment by rvernica** [comment]
In general, the operators in SciDB-Py follow the prototype of the operators in SciDB. [store](https://paradigm4.atlassian.net/wiki/spaces/ESD169/pages/50856164/store) does not have a `temp` parameter in SciDB. Moreover, your example maps directly to `store(build(...), ...)` so there is no `create array` issued.

We could have a special case if the operator is `store` and it has the `temp` argument. In such cases we could issue a `create array` call first, but we don't know the schema. We would have to pick up the schema from the previous operators, if any is present. This might be tricky.

Now, this might not be what you are looking for but, if an array name is not specified for the `store` operator, an array name is generated. Arrays with generated names are removed when the returned Array object is garbage collected. This behavior can be changed by specifying the gc=False argument to the store operator.

**Comment by rvernica** [comment]
We should support both:
```python
>>> db.build('<x:int8 not null>[i=0:2]', 'i + 10').store('foo', temp=True)
```
and
```python
>>> db.build('<x:int8 not null>[i=0:2]', 'i + 10').store(temp=True)
```

**Comment by rvernica** [comment]
It's tricky to make this work for cases like:
```python
>>> db.input(upload_data=numpy.arange(3)).store('foo', temp=True)
```
because the input schema, file name ,and file format are determined right before the query is executed in the `db.iquery` function and we cannot do a `show` call inside the `store` function.

**Comment by rvernica** [comment]
Complex `input` operators are not supported:
```python
>>> db.input(upload_data=numpy.arange(3)).store('foo', temp=True)
HTTPError: "temp=True" not supported for complex queries
406 Client Error: UserQueryException in file: src/query/parser/Driver.cpp function: fail line: 148
Error id: scidb::SCIDB_SE_PARSER::SCIDB_LE_QUERY_PARSING_ERROR
Error description: Error during query parsing. Query parser failed with error 'syntax error'.
input({sch}, '{fn}', 0, '{fmt}')
```
Notice the *"temp=True" not supported for complex queries* statement in the exception, followed by the actual exception (for corner-cases). Simple `input` operators are supported:
```python
>>> db.input('<x:int64>[i=0:2]', '/tmp/1').store('foo', temp=True)
Array(DB('http://localhost:8080', None, None, None, None), 'foo')
>>> db.list()[['name', 'temporary']]
    name  temporary
0   foo       True
```

**Comment by apoliakov** [comment]
Fair enough, thanks Rares. I'll talk to core dev about maybe adding "temp=true" to AFL store.
